galasa-ui/src/tests/usersTable.test.tsx

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+import React from 'react';
+import { render, screen, waitFor } from '@testing-library/react';
+import UsersTable from '@/components/users/UsersTable';
+import { UserData } from '@/generated/galasaapi';
+import '@testing-library/jest-dom';
+
+// ------------------------------
+// Mocks
+// ------------------------------
+
+// Mock Carbon's Loading component to render a simple div.
+jest.mock('@carbon/react', () => {
+  const originalModule = jest.requireActual('@carbon/react');
+  return {
+    __esModule: true,
+    ...originalModule,
+    Loading: () => <div data-testid="loading">Loading...</div>
+  };
+});
+
+// Mock the ErrorPage component.
+jest.mock('@/app/error/page', () => {
+  function ErrorPageMock() {
+    return <div data-testid="error-page">Error Page</div>;
+  }
+  return ErrorPageMock;
+});
```

Comment:
I don't think we need to mock the error page - we can instead check that the error page was rendered